climakitae/util/warming_levels.py

```diff
@@ -0,0 +1,233 @@
+
+    return warming_data
+
+
+def _get_sliced_data(y, level, gwl_times, months, window):
+    """Calculate warming level anomalies.
+    Warming level is computed for each individual simulation/scenario.
+
+    Parameters
+    ----------
+    y: xr.DataArray
+        Data to compute warming level anomolies, one simulation at a time via groupby
+    gwl_times: pd.DataFrame
+        Global warming levels table indicating when each unique model/run/scenario (simulation) reaches each warming level.
+    level: float
+        Warming level. Must be a valid column in gwl_times table.
+    months: list of int
+        Months of the year (in integers) to compute function for.
+        i.e. for a full year, months = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]
+    window: int
+        Years around Global Warming Level (+/-) \n (e.g. 15 means a 30yr window)
+
+    Returns
+    --------
+    anomaly_da: xr.DataArray
+    """
+    # Get the years when the global warming level is reached for all levels available in the gwl_times dataframe
+    gwl_times_subset = gwl_times.loc[_extract_string_identifiers(y)]
+
+    # Checking if the centered year is null, if so, return dummy DataArray
+    center_time = gwl_times_subset.loc[str(float(level))]
+
+    # Dropping leap days before slicing time dimension because the window size can affect number of leap days per slice
+    y = y.loc[~((y.time.dt.month == 2) & (y.time.dt.day == 29))]
+
+    if not pd.isna(center_time):
+
+        # Find the centered year
+        centered_year = pd.to_datetime(center_time).year
+        start_year = centered_year - window
+        end_year = centered_year + (window - 1)
+
+        sliced = y.sel(time=slice(str(start_year), str(end_year)))
+
+        # Creating a mask for timestamps that are within the desired months
+        valid_months_mask = sliced.time.dt.month.isin([months])
+
+        # Resetting and renaming time index for each data array so they can overlap and save storage space.
+        expected_counts = {
+            "monthly": window * 2 * 12,
+            "daily": window * 2 * 365,
+            "hourly": window * 2 * 8760,
+        }
+        # There may be missing time for time slices that exceed the 2100 year bound. If that is the case, only return a warming slice for the amount of valid data available AND correctly center `time_from_center` values.
+        # Otherwise, if no time is missing, then the warming slice will just center the center year.
+        sliced["time"] = np.arange(
+            -expected_counts[y.frequency] / 2,
+            expected_counts[y.frequency] / 2
+            - (expected_counts[y.frequency] - len(sliced)),
+        )
+
+        # Removing data not in the desired months (in this new time dimension)
+        sliced = sliced.sel(time=valid_months_mask)
+
+        # Assigning `centered_year` as a coordinate to the DataArray
+        sliced = sliced.assign_coords({"centered_year": centered_year})
+
+    else:
```

Comment:
I know this was not code that was related to this PR, but could you add a comment above/below this else that describes why this is done?

i.e.

```
# This clause creates an empty DataArray with similar shape to real WL slices to get dropped after the `.groupby` method is finished.
```

Reply:
@claalmve do you mind adding that, or putting the comment here in GitHub so I can add? I'm actually not super sure what this call does (I copied your code here) 

Thanks!

Reply:
Wait sorry I see that you included an example of the code comment already hahaha thanks!